packages/back-end/src/models/InformationSchemaTablesModel.ts

```diff
@@ -0,0 +1,72 @@
+import omit from "lodash/omit";
+import z from "zod";
+import mongoose from "mongoose";
+import { InformationSchemaTablesInterface } from "../types/Integration";
+import { errorStringFromZodResult } from "../util/validation";
+import { logger } from "../util/logger";
+import { usingFileConfig } from "../init/config";
+
+const informationSchemaTablesSchema = new mongoose.Schema({
+  id: String,
+  organization: {
+    type: String,
+    index: true,
+  },
+  tableName: String,
+  tableSchema: String,
+  databaseName: String,
+  columns: {
+    type: [Object],
+    required: true,
+    validate: {
+      validator(value: unknown) {
+        const zodSchema = z.array(
+          z.object({
+            columnName: z.string(),
+            path: z.string(),
+            dataType: z.string(),
+          })
+        );
+
+        const result = zodSchema.safeParse(value);
+
+        if (!result.success) {
+          const errorString = errorStringFromZodResult(result);
+          logger.error(errorString, "Invalid Columns name");
+        }
+
+        return result.success;
+      },
+    },
+  },
+  dateCreated: Date,
+  dateUpdated: Date,
+});
+
+type InformationSchemaTablesDocument = mongoose.Document &
+  InformationSchemaTablesInterface;
+
+const InformationSchemaTablesModel = mongoose.model<InformationSchemaTablesDocument>(
+  "InformationSchemaTables",
+  informationSchemaTablesSchema
+);
+
+/**
+ * Convert the Mongo document to an InformationSourceInterface, omitting Mongo default fields __v, _id
+ * @param doc
+ */
+const toInterface = (
+  doc: InformationSchemaTablesDocument
+): InformationSchemaTablesInterface => omit(doc.toJSON(), ["__v", "_id"]);
+
+export async function createInformationSchemaTables(
+  tables: InformationSchemaTablesInterface[]
+): Promise<InformationSchemaTablesInterface[]> {
+  if (usingFileConfig()) {
+    throw new Error("Cannot add. Data sources managed by config.yml");
+  }
```

Comment:
Same here. Either remove this check or add a comment that we want to remove it eventually.

Reply:
Commented added, and task added to Linear project.